src/cli.rs

```diff
@@ -157,36 +143,65 @@ pub enum IotHubDeviceUpdateConfig {
         /// optional: generate bmap file
         #[arg(short = 'b', long = "generate-bmap-file")]
         generate_bmap: bool,
+        /// optional: pack image [xz, bzip2, gzip]
+        #[arg(short = 'p', long = "pack-image", value_enum)]
+        compress_image: Option<Compression>,
     },
 }
 
 #[derive(Parser, Debug)]
 #[command(after_help = COPYRIGHT)]
-/// establish ssh connections to devices
-pub struct SshConfig {
-    /// username for the login on the device.
-    #[arg(short = 'u', long = "user", default_value = "omnect")]
-    pub username: String,
-    /// optional: path where the ssh key pair, the certificates, and the
-    /// temporary ssh configuration is stored. Defaults to system local data
-    /// directories (e.g. ${XDG_RUNTIME_DIR}/omnect-cli on Linux).
-    #[arg(short = 'd', long = "dir")]
-    pub dir: Option<std::path::PathBuf>,
-    /// optional: path to a pre-existing ssh private key that is used. Note:
-    /// this expects the existence of a corresponding <key-path>.pub file.
-    /// If not specified, omnect-cli creates a key pair for this connection.
-    #[arg(short = 'k', long = "key")]
-    pub priv_key_path: Option<std::path::PathBuf>,
-    /// optional: path where the ssh configuration is stored. Defaults to system
-    /// local data directories (e.g. ${XDG_RUNTIME_DIR}/omnect-cli/ssh_config on
-    /// Linux).
-    #[arg(short = 'c', long = "config-path")]
-    pub config_path: Option<std::path::PathBuf>,
-    /// address of the backend API
-    #[arg(short = 'b', long = "backend", default_value = "https://cp.omnect.conplement.cloud")]
-    pub backend: String,
-    /// name of the device for which the ssh tunnel should be created.
-    pub device: String,
+/// file handling
+pub enum Ssh {
```

Comment:
Why not `SshConfig` all the other root structs are imho named *Config.
What is the difference here?

Reply:
imho the `Connection` subcommand doesn't fit to `SshConfig`

Reply:
that creates a `ssh_config` file we use for connecting, e.g. via
` ssh -F /home/marcel/.local/share/omnect-cli/ssh_config test_device`

Reply:
Ok maybe `Ssh --> SshConfig` and `Connection --> SetConnection`?